apps/backend/src/app/modules/users/users.service.ts

```diff
@@ -480,6 +480,34 @@ export class UsersService {
     return DtoFactory(FollowDto, dbResponse);
   }
 
+  async followUsers(userID: number, usersIDs: number[]): Promise<FollowDto[]> {
+    const users = await this.db.user.findMany({
+      where: { id: { in: usersIDs } }
+    });
+
+    if (users.length !== usersIDs.length)
+      throw new NotFoundException('One or more users not found');
+
+    const follows = await Promise.all(
+      users.map(async (user) => {
+        if (user.id === userID)
+          throw new BadRequestException('User cannot follow themselves');
+
+        const isFollowing = await this.getFollower(userID, user.id);
+        if (isFollowing)
+          throw new BadRequestException(
+            'User is already following one or more of the target users'
+          );
+
+        return this.db.follow.create({
+          data: { followeeID: userID, followedID: user.id }
+        });
+      })
+    );
+
+    return follows.map((follow) => DtoFactory(FollowDto, follow));
+  }
+
```

Comment:
Sorry I must've forgotten to flag this -- you have the same issue here as with my previous comment, doing individual queries for each follow.

The "already following" thing is probably because Prisma will throw if a Follow record already exists, but in that case you could first do a query to check where the user already follows any of the current, and throw if they do.

The self-following thing can done with just `if (userIDs.includes(userID)) throw new ... `